addons/website_sale/models/website_snippet_filter.py

```diff
@@ -167,7 +167,8 @@ def _get_products_recently_sold_with(
         self, website, limit, domain, product_template_id, **kwargs,
     ):
         products = self.env['product.product']
-        current_template = self.env['product.template'].browse(int(product_template_id)).exists()
+        current_template = product_template_id \
+            and self.env['product.template'].browse(int(product_template_id)).exists()
```

Comment:
```suggestion
        current_template = self.env['product.template'].browse(
            product_template_id and int(product_template_id)
        ).exists()
```
Since `browse` supports a falsy value to return an empty recordset, this will ensure that `current_template` is always a recordset (though sometimes empty) and that it's type is consistent.